lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -135,9 +136,11 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
           return signalPassFailure();
 
       // Record all the RefType ports to be removed later.
-      for (size_t portNum = 0, e = module.getNumPorts(); portNum < e; ++portNum)
+      size_t numPorts = module.getNumPorts();
+      refPortsToRemoveMap[module].resize(numPorts);
+      for (size_t portNum = 0; portNum < numPorts; ++portNum)
         if (module.getPortType(portNum).isa<RefType>())
-          refPortsToRemoveMap[module].push_back(portNum);
```

Comment:
It may be important/desirable to only create this entry if there are RefType's present, in which case it needs to be sized to have bits for all ports.  Or create the BitVector in this loop and only add it to the map if anything set?

Reply:
Sounds good. I went with moving the resizing operation inside of the `if()` statement, which should only implicitly create the entry when I first access it with the `[]` operator. Does this look good to you? https://github.com/llvm/circt/pull/3837/commits/b278e15b1cc4437d64118c72e53802bef4d1c732